mitodl/mitxpro
# mitxpro-1393 Add contract number to b2b order

#### Pre-Flight checklist


#### What are the relevant tickets?
Fixes #1393 

#### What's this PR do?

- [x]  Add an optional "contract number" field to the form
- [x]  Allow the value to be set by a query string in the URL, i.e. ?contract=123456789
- [x]  the field should only be visible when the query string is set, i.e. users without a contract number won't see it.
- [x]  include the contract number in the receipts (email and HTML)
- [x]  include the contract number as a merchant defined field in Cybersource

#### How should this be manually tested?

- Goto /ecommerce/bulk
- Try contract_number in query string i.e. ?contract_number=123456789
- You should see a new filed `Contract Number`, containing value from query string
- Field should be saved upon submission

#### Screenshots (if appropriate)
**Receipt page:**
<img width="1063" alt="Screenshot 2020-01-07 at 1 31 36 PM" src="https://user-images.githubusercontent.com/4245618/71966689-de1e6700-3223-11ea-8be6-0e9f15050371.png">

**Recept email:**
<img width="712" alt="71365635-6a445380-25c1-11ea-9f65-0aa8331cd6e1" src="https://user-images.githubusercontent.com/4245618/71966712-e9719280-3223-11ea-9971-a214e43254c4.png">
